pkg/console/operator/sync_v400.go

```diff
@@ -332,10 +332,10 @@ func SyncRoute(co *consoleOperator, operatorConfig *operatorv1.Console) (*routev
 
 	// we will not proceed until the route is valid. this eliminates complexity with the
 	// configmap, secret & oauth client as they can be certain they have a host if we pass this point.
-	if len(rt.Spec.Host) == 0 {
-		// TODO STATUS
+	_, err := routesub.GetCanonicalHost(rt)
+	if err != nil {
```

Comment:
Not new, but I think `GetCanonicalHost` should just return `nil` when there's no host set yet other than an error. I would try to distinguish between errors from the API and something just not being set yet.

Reply:
agree, i'll update.